### ISSUE TITLE
Exercise #5: Maybe Level 5 - or()

### ISSUE THREAD

**Issue by anniedaang** [nus-cs2030/2223-s1#293]
I need help with the method or() on level 5 in the self exercise 5.

_"another variant is the or method which takes in a Supplier of a Maybe<T> as the alternative. If a value does not exist in Maybe, the alternative Maybe produced by the supplier is returned."_ 
Below is my code: I don't know how to cast the Supplier or the return value for this to work.

<img width="417" alt="Skärmavbild 2022-11-20 kl  16 49 35" src="https://user-images.githubusercontent.com/71130637/202893459-121a2d50-3458-413e-aedd-9a9a7fb729bb.png">

**this is the error message that I receive:**

<img width="572" alt="Skärmavbild 2022-11-20 kl  16 50 59" src="https://user-images.githubusercontent.com/71130637/202893503-d2b42d8a-ed41-4e9f-a253-5d813a9b2b4f.png">

I would really appreciate it if someone could give me some guidance or explain the logic.

**Comment by jamielfh** [comment]
Hi there! I think the issue lies in the difference between the variable mo, which is typed as a Maybe<Object>, and the Maybe you will get from the Supplier, which is a Maybe<Integer>. Thus, you will need to find a way to convert / re-type the result of the Supplier's get() function to a Maybe<Object>.

What I did was to take in a Supplier that could return results belonging to child classes of Object using Supplier<? extends Maybe<? extends T>>. Within the if clause of the or method, I first casted supplier.get() to a Maybe<? extends T>, then retrieving its thing and returning it in a Maybe<T> object. Hope this helps!
